cbrain_task/spark_handler/bourreau/spark_handler.rb

```diff
@@ -35,6 +35,34 @@ def self.override_save_results #:nodoc:
   # # See the CbrainTask Programmer Guide
   def save_results #:nodoc:
     params    = self.params
+
+    # No matter how many errors occur, we need to save as many output
+    # files as possible and carry the error state to the end.
+    succeeded = true
+
+    # Make sure SPARKstage[*]of3 completed successfully by checking its exit status
+    # in +exit_cluster_filename+.
+    if !File.exists?(exit_cluster_filename)
```

Comment:
This seems like a really complex set of if/else/ etc, given that at the end you just return false as soon as succeeded == false

Reply:
You could simply have three sections in the form:

```ruby
if condition
  addlog error message
  return false
end
```

Reply:
That way you don't even need the variable `succeeded`.